frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import {
+  Button,
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  Label,
+  LabelGroup,
+  Modal,
+  TextInput,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import DashboardDescriptionListGroup, {
+  DashboardDescriptionListGroupProps,
+} from './DashboardDescriptionListGroup';
+
+type EditableTextDescriptionListGroupProps = Partial<
+  Pick<DashboardDescriptionListGroupProps, 'title' | 'contentWhenEmpty'>
+> & {
+  labels: string[];
+  saveEditedLabels: (labels: string[]) => Promise<void>;
+};
+
+const EditableLabelsDescriptionListGroup: React.FC<EditableTextDescriptionListGroupProps> = ({
+  title = 'Labels',
+  contentWhenEmpty = 'No labels',
+  labels,
+  saveEditedLabels,
+}) => {
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [unsavedLabels, setUnsavedLabels] = React.useState(labels);
+  const [isSavingEdits, setIsSavingEdits] = React.useState(false);
+
+  const editUnsavedLabel = (newText: string, index: number) => {
```

Comment:
I feel that if this is gonna be reused in the details view we can move it into a utils right?
Not needed right now, can be refactor on @dpanshug pr

Reply:
Resolving conversation it can be done in another PR